### PR TITLE
Auto associations remove foreign keys

### PR DESCRIPTION
**Pull request by Fridus**
The option `associations` can to be a boolean or an object.
```
        rest.resource({
          model: test.models.User,
          endpoints: ['/usersWithoutFK', '/usersWithoutFK/:id'],
          associations: {
            removeForeignKeys: true
          }
        });
```

### CODE REVIEW

File: lib/Resource.js [hunk 1]
```diff
@@ -38,7 +38,14 @@ var Resource = function(options) {
   this.search = options.search;
   this.sort = options.sort;
 
+  this.associationOptions = {
```
Comment (mbroadst) [inline lib/Resource.js]: this should be added to the ```mergeDefaults``` above, below ```include```. Then you don't have to use _.extend, and we can more easily track all the defaults for documentation purposes

File: lib/Resource.js [hunk 2]
```diff
@@ -38,7 +38,14 @@ var Resource = function(options) {
...
+    removeForeignKeys: false
+  };
+
   if (options.associations) {
```
Comment (mbroadst) [inline lib/Resource.js]: I think this should be ```!!options.associations``` so we consider all potentially falsey values (now that options.associations can be an object or a boolean)

Comment (Fridus) [inline lib/Resource.js]: @mbroadst Ok ok

File: lib/Resource.js (hunk 1 above)
Comment (Fridus) [inline lib/Resource.js]: @mbroadst If it is added to the object `options` (L12) then the option `associations: true` will remove the default values. No ?

Comment (mbroadst) [inline lib/Resource.js]: Oh that's right... hm. Okay well I guess this is the way to go then

**Comment by Fridus** [comment]
@mbroadst Thanks. Please, could you rebase `auto-associations` on master ? :)

**Comment by mbroadst** [comment]
@Fridus done :smile:

**Comment by Fridus** [comment]
@mbroadst Thanks ! Why when you rebase, marge commits are lost ? This is your choice ?

**Comment by mbroadst** [comment]
@Fridus yeah, actually they shouldn't go missing (maybe that's a github-ism), but I do a ```git pull --rebase origin master``` on that branch to keep the history clean (since auto-associations is guaranteed to be based off the master repo)

**Pull request merged**